Add Error Features for JS modules

andistorm opened the pull request:
Add Error Handling/Reporting funcionality to JS framework

* Raise error: mod.provides.<impl_id>.raise.<error_namespace>_<error_name>('<message>', '<severity')
* Request clear error by uuid: mod.provides.<impl_id>.request_clear_uuid('<uuid>')
* Request clear all error of one type: mod.provides.<impl_id>.request_clear_all_of_type.<error_namespace>_<error_name>()
* Request clear all all of module/impl: mod.provides.<impl_id>.request_clear_all()
* Subscribe specific error type: setup.uses.<req_id>.subscribe_error.<error_namespace>_<error_name>(subscribe_error_cb, subscribe_error_cleared_cb)
* Subscribe all errors of req: setup.uses.<req_id>.susbcribe_all_errors(susbcribe_error_cb, subscribe_error_cleared_cb)

Limitations:
* As it is not possible to register multiple handlers to subscribed var, it is not possible to subscribe multiple times to an error type
---------------------------------------------------------------------------
Fix Everest::error::Severity
* Mid -> Medium
* Do not ignore set severity, but use it in Error constructor

andistorm committed:
Fix Everest::error::Severity

* Mid -> Medium
* Do not ignore set severity, but use it in Error constructor

Signed-off-by: Andreas Heinrich <andreas.heinrich@rwth-aachen.de>

andistorm committed:
Add Error Handling/Reporting funcionality to JS framework

* Raise error: mod.provides.<impl_id>.raise.<error_namespace>_<error_name>('<message>', '<severity')
* Request clear error by uuid: mod.provides.<impl_id>.request_clear_uuid('<uuid>')
* Request clear all error of one type: mod.provides.<impl_id>.request_clear_all_of_type.<error_namespace>_<error_name>()
* Request clear all all of module/impl: mod.provides.<impl_id>.request_clear_all()
* Subscribe specific error type: setup.uses.<req_id>.subscribe_error.<error_namespace>_<error_name>(subscribe_error_cb, subscribe_error_cleared_cb)
* Subscribe all errors of req: setup.uses.<req_id>.susbcribe_all_errors(susbcribe_error_cb, subscribe_error_cleared_cb)

Limitations:
* As it is not possible to register multiple handlers to subscribed var, it is not possible to subscribe multiple times to an error type

Signed-off-by: Andreas Heinrich <andreas.heinrich@rwth-aachen.de>

hikinggrass commented on everestjs/everestjs.cpp:
```diff
@@ -380,17 +473,75 @@ static Napi::Value boot_module(const Napi::CallbackInfo& info) {
+                    for (const auto& error_name_it : error_namespace_it.value().items()) {
+                        const std::string& error_name = error_name_it.key();
+                        impl_raise_errors_prop.DefineProperty(Napi::PropertyDescriptor::Value(
+                            error_namespace + "_" + error_name,
```
This is used a few times in the loop, could be defined as a variable

andistorm committed:
Improve readability

Use error_type and prop_key instead of combination of error_namespace and error_name

Signed-off-by: Andreas Heinrich <andreas.heinrich@rwth-aachen.de>
